### PR TITLE
Azsnpvtpm: Replace anyhow error crate with thiserror crate

### PR DESCRIPTION
**Pull request by kartikjoshi21**

### CODE REVIEW

File: attestation-service/verifier/Cargo.toml [hunk 1]
```diff
@@ -30,6 +31,7 @@ csv-rs = { git = "https://github.com/openanolis/csv-rs", rev = "b74aa8c", option
 eventlog-rs = { version = "0.1.3", optional = true }
 hex.workspace = true
 jsonwebtoken = { workspace = true, default-features = false, optional = true }
+jsonwebkey = "0.3.5"
```
Comment (mkulke) [inline attestation-service/verifier/Cargo.toml:34]: we should share the jsonwebtoken crate in the workspace

File: attestation-service/verifier/src/az_snp_vtpm/mod.rs [hunk 2]
```diff
@@ -129,13 +153,13 @@ fn verify_snp_report(
...
     vendor_certs: &VendorCertificates,
-) -> Result<()> {
+) -> Result<(), CertError> {
     let vcek_data = vcek.0.to_der().context("Failed to get raw VCEK data")?;
     let cert_chain = [CertTableEntry::new(CertType::VCEK, vcek_data)];
     verify_report_signature(snp_report, &cert_chain, vendor_certs)?;
 
     if snp_report.vmpl != HCL_VMPL_VALUE {
-        bail!("VMPL of SNP report is not {HCL_VMPL_VALUE}");
+        return Err(CertError::VmplIncorrect(HCL_VMPL_VALUE.to_string()));
```
Comment (mkulke) [inline attestation-service/verifier/src/az_snp_vtpm/mod.rs:162]: why do we have to convert to string?

File: attestation-service/verifier/src/az_snp_vtpm/mod.rs [hunk 3]
```diff
@@ -33,10 +36,28 @@ pub struct AzSnpVtpm {
...
+    LoadMilanCert,
+    #[error("TPM quote nonce doesn't match expected report_data")]
+    NonceMismatch,
+    #[error("SNP report report_data mismatch")]
+    SnpReportMismatch,
+    #[error("VMPL of SNP report is not {0}")]
+    VmplIncorrect(String),
+    #[error("Quote error")]
+    Quote(#[from] QuoteError),
+    #[error("jsonewebkey conversion error")]
```
Comment (mkulke) [inline attestation-service/verifier/src/az_snp_vtpm/mod.rs:51]: typo:

```suggestion
    #[error("JWK conversion failed")]
```

**Review by mkulke — COMMENTED** [review]
few nits, otherwise looks good.

File: attestation-service/verifier/Cargo.toml (hunk 1 above)
Comment (kartikjoshi21) [reply attestation-service/verifier/Cargo.toml:34]: jsonwebtoken  or jsonwebkey ?

Comment (mkulke) [reply attestation-service/verifier/Cargo.toml:34]: ah, jsonwebkey. but do we need to add this dependency in this PR at all?

Comment (kartikjoshi21) [reply attestation-service/verifier/Cargo.toml]: Its not required now, this dependency already included and merged in other PR

**Review by jialez0 — APPROVED** [review]

**Review by jialez0 — APPROVED** [review]

**Pull request merged**